src/quantum/tp6.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { withStyles } from '@material-ui/core';
+import PropTypes from 'prop-types';
+import _ from '../utils/more_lodash';
+import { TP6_PAGES } from './config';
+import DashboardPage from '../components/DashboardPage/index';
+import PerfherderGraphContainer from '../containers/PerfherderGraphContainer/index';
+
+
+const styles = {
+  // This div helps with canvas size changes
+  // https://www.chartjs.org/docs/latest/general/responsive.html#important-note
+  body: {
+    backgroundColor: 'white',
+  },
+  area: {
+    display: 'flex',
+  },
+  column: {
+    flex: '50%',
+    alignItems: 'center',
+  },
+  chart: {
+    justifyContent: 'center',
+    padding: '1rem',
+  },
+};
+
+class TP6 extends React.Component {
+  constructor(props) {
+    super(props);
+    const { classes, match } = this.props;
+    const bits = match.params.bits;
+
+    this.state = {
+      allCharts: _
+        .chain(TP6_PAGES)
+        .filter(row => row.bits === bits)
+        // GROUP BY title
+        .groupBy(row => row.title)
+        // LOOP OVER EACH KEY/VALUE
+        .toPairs()
+        .map(([title, series], i) => (
+          <div key={`page_${bits}_${i}`} className={classes.chart}>
+            <PerfherderGraphContainer
```

Comment:
Please don't put UI elements withing the state.

Instead create an object that represents your UX.
Remember "Your UI is a representation of your state".

Reply:
@helfi92 can I generate the React elements in the constructor, rather than passing them in the props?

Reply:
I'm not sure, you could try. However, I would rather create them in a different file and import them directly. React constructors are not typically used for creating React elements.